screens/Home.tsx

```diff
@@ -1,19 +1,122 @@
-import React from 'react';
-import { StyleSheet, Text, View } from 'react-native';
+import React, { useEffect, useState } from 'react';
+import {
+  View,
+  StyleSheet,
+  FlatList,
+  ActivityIndicator,
+} from 'react-native';
 
 
-export default function HomeScreen() {
+import { instance as axiosInstance } from '../helpers/axiosConfig';
+import { StackScreenProps } from '@react-navigation/stack';
+import { HomeStackParamList } from '../Root';
+import Submission from '../types/Submission';
+import PaginationData from '../types/PaginationData';
+import SubmissionTableItem from '../components/SubmissionTableItem';
+
+type GetSubmissionsResponse = {
+  data: Submission[],
+  pagination: PaginationData,
+}
+
+export default function HomeScreen({ route, navigation }: StackScreenProps<HomeStackParamList, 'Home'>) {
+  const [data, setData] = useState<Submission[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [page, setPage] = useState<number>(1);
+  const [isRefreshing, setIsRefreshing] = useState<boolean>(false);
+  const [isFetching, setIsFetching] = useState<boolean>(false);
+  const [didFetchAllPages, setDidFetchAllPages] = useState<boolean>(false);
+
+  useEffect(() => {
+    fetchSubmissions();
+  }, [page]);
+
+  useEffect(() => {
```

Comment:
ok, this is getting out of hand, let's refactor this in the next PRs into react query